providers/oidc.go

```diff
@@ -71,21 +74,25 @@ func (p *OIDCProvider) Redeem(ctx context.Context, redirectURL, code string) (s
 
 // RefreshSessionIfNeeded checks if the session has expired and uses the
 // RefreshToken to fetch a new Access Token (and optional ID token) if required
-func (p *OIDCProvider) RefreshSessionIfNeeded(ctx context.Context, s *sessions.SessionState) (bool, error) {
-	if s == nil || (s.ExpiresOn != nil && s.ExpiresOn.After(time.Now())) || s.RefreshToken == "" {
+func (p *OIDCProvider) RefreshSessionIfNeeded(ctx context.Context, ps mw.ProxyState, s *sessions.SessionState) (bool, error) {
```

Comment:
I think we can simplify the implementation of this moving forward:

What if we switch the provider interface to `RefreshSession`.  Calling this method would do no timing verifications, it would just refresh a session if called.

Then we can centralize all the timing logic on the refresh in the middleware: https://github.com/oauth2-proxy/oauth2-proxy/blob/cc6b808c1762538cc6e7c5814a9a33b5ee9ae63d/pkg/middleware/stored_session.go#L115

Then I think we can remove needing to pass around the `ProxyState` into all the provider methods?

Reply:
I wanted to preserve the backwards compatibility with existing functionality for people who use it as is. TTL on cookie refresh makes zero sense to me altogether, but I didn't want to get rid of it entirely in case someone had legitimate reasons for having that behavior.

Reply:
The next release (v7.0.0) is breaking.  If it makes sense for the long term maintainability & architecture of the project; now is the time to break stuff and ignore backwards compatibility.

We don't get these opportunities to clean up the legacy cruft often 😄

Reply:
I REALLY want to make this `RefreshSession` if you didn't pick up that issue, that was the direction I was going to go if I picked it up 🤣 

We've had a few PRs recently where we went back and forth on what to do in these `RefreshSessionIfNeeded` methods and we ended up preserving the redundant time check logic in all of them just to keep everything consistent -- even though it was likely the wrong decision and DRYing up this logic is the right direction.

Reply:
Ok, breaking changes it is. :smile: